xcm/xcm-builder/src/pay.rs

```diff
@@ -61,20 +53,22 @@ impl<
 		asset_kind: Self::AssetKind,
 		amount: Self::Balance,
 	) -> Result<Self::Id, ()> {
-		let beneficiary = AccountIdConverter::reverse(who.clone()).map_err(|_| ())?;
+		let (dest, maybe_sender) = Sender::get().split_last_interior();
+		let sender = maybe_sender.map(|s| Junctions::X1(s)).ok_or(())?;
 		let mut message = Xcm(vec![
 			UnpaidExecution { weight_limit: Unlimited, check_origin: None },
+			DescendOrigin(sender),
```

Comment:
without this, we just trying to transfer from `MultiLocation` `Here` to `beneficiary`, and it will fail